deploy/New-IdentityProvider.ps1

```diff
@@ -0,0 +1,90 @@
+<#
+.Description
+This allows you to create an Identity Provider for providing tokens
+to an application - this is optional.
+#> 
+
+[CmdletBinding()]
+param(
+    # The name of the AAD Client Application
+    [Parameter(Mandatory=$false)]
+    [string] $AADClientAppName="ccf_demo_client",
+    # The name of the AAD API Application
+    [Parameter(Mandatory=$false)]
+    [string] $AADApiAppName="ccf_demo_api",
+    # Tenant where the Application Registrations are
+    [Parameter(Mandatory=$true)]
+    [string] $TenantId
+)
+
+# The Magic Strings are well known Ids
+$MSGraphAppId="00000003-0000-0000-c000-000000000000"
+# $Scope_OpenID="37f7f235-527c-4136-accd-4a02d197296e"
+# $Scope_OfflineAccess="7427e0e9-2fba-42fe-b0c0-848c9e6a8182"
+$Scope_UserRead="e1fe6dd8-ba31-4d61-89e7-88639da4683d"
+
+# Hold all of our keys in a hashtable
+$output = @{}
+# Clean our .env file
+Remove-Item ../.env -Force -ErrorAction SilentlyContinue
+
+# Connect with the correct scopes that you need to perform the management operations.
+# These are not the scopes you are granting to the applications.
+$RequiredScopes = @("Directory.AccessAsUser.All", "Directory.ReadWrite.All", "Application.ReadWrite.All")
+Connect-MgGraph -TenantId $TenantId -Scopes $RequiredScopes
+
+# 1. Create the application that secures the API.
+# https://learn.microsoft.com/en-us/powershell/module/microsoft.graph.applications/new-mgapplication?view=graph-powershell-1.0
+#$userImpersonationScopeId=New-Guid
+$api_app_manifest = @{
+    DisplayName = $AADApiAppName
+    SignInAudience = "AzureADMyOrg"
+    RequiredResourceAccess = @{
+        ResourceAppId = $MSGraphAppId
+        ResourceAccess = @(
+            @{
+                Id = $Scope_UserRead
+                Type = "Scope"
+            })
+        }
+}
+$api_app = New-MgApplication @api_app_manifest
```

Comment:
your parameters and the module's parameters for the function calls are `PascalCase`. Your `$` parameters (I don't know if this shell specific), are not. Was this intentional or even required?

Reply:
Done - many thanks!